tests/functional/test_install_config.py

```diff
@@ -209,3 +216,61 @@ def test_install_no_binary_via_config_disables_cached_wheels(
     assert "Building wheel for upper" not in str(res), str(res)
     # Must have used source, not a cached wheel to install upper.
     assert "Running setup.py install for upper" in str(res), str(res)
+
+
+def test_prompt_for_authentication(script, data, cert_factory):
+    """Test behaviour while installing from a index url
+    requiring authentication
+    """
+    cert_path = cert_factory()
+    ctx = ssl.SSLContext(ssl.PROTOCOL_SSLv23)
+    ctx.load_cert_chain(cert_path, cert_path)
+    ctx.load_verify_locations(cafile=cert_path)
+    ctx.verify_mode = ssl.CERT_REQUIRED
+
+    server = make_mock_server(ssl_context=ctx)
+    server.mock.side_effect = [
+        package_page({
+            "simple-3.0.tar.gz": "/files/simple-3.0.tar.gz",
+        }),
+        authorization_response(str(data.packages / "simple-3.0.tar.gz")),
+    ]
+
+    url = "https://{}:{}/simple".format(server.host, server.port)
+
+    with server_running(server):
+        result = script.pip('install', "--index-url", url,
+                            "--cert", cert_path, "--client-cert", cert_path,
+                            'simple', expect_error=True)
+    print(result)
```

Comment:
Why is there a `print` here ?

Reply:
Oops, my bad. Fixed in https://github.com/pypa/pip/pull/8182/commits/4ecd7ecbc705db7ad9953b1cab2e3b8a15c9d748